### ISSUE TITLE
fish compatibility

### ISSUE THREAD

**Issue by curio77** [denisidoro/navi#171]
**Describe the bug**
In fish, navi works as expected for some variables in commands, while for others, it just inserts `query` instead of prompting for a selection.  It is not clear to me what the discriminating factor is.

**To Reproduce**
1. Set up navi in fish as described.
2. Open navi via Ctrl-G.
3. From the default cheats, select the `docker-compose up -d <service_name>` cheat that ships with navi.
4. See that this gets expanded into `docker-compose up -d query`.

**Expected behavior**
navi should prompt for a service name.

**Versions:**
 - OS: Linux (Gentoo, but shouldn't matter)
 - fish v3.0.2

**Comment by denisidoro** [comment]
What does `navi --version` output?

**Comment by denisidoro** [comment]
Does this work when using bash, for instance?

**Comment by curio77** [comment]
It's up-to-date (GitHub HEAD), v0.17.0.

**Comment by curio77** [comment]
In Bash everything works fine.

**Comment by denisidoro** [comment]
Could you please check the behavior when calling `navi --no-autocomplete`?

**Comment by denisidoro** [comment]
There was a typo on my previous message. I've fixed it

**Comment by curio77** [comment]
Then I do get an input prompt.  So one cause appears to be having input-only variables (that would auto-select to none).

However, in another case, I observe the same effect (artifical example):

```
# Test
true <foo>

$ foo: echo 1 2 3 | tr " " "\n"
```

Here, I too get that `query` inserted (yielding `true query`), even though there's a list (1, 2, and 3) to select from.  With `--no-auto-select`, navi returns to the command line without filling in anything (command is blank). In bash, I get the expected selection list.

**Comment by denisidoro** [comment]
Hi, @curio77. Sorry for the extremely late response. I was on vacations.

I'm not able to reproduce the problem. Using `fish 3.0.2` on `OSX` I am able to select the values accordingly.

Do you still face this issue in the latest version? I know I'm asking too much, but maybe you could find a minimal docker image that has this problem?

**Comment by denisidoro** [comment]
I've run this in a docker container and wasn't able to reproduce as well: https://www.katacoda.com/denisidoro/scenarios/navi:
```bash
cd "$(navi home)/cheats"
vi test.cheat # and paste the code below
cd
echo foo > "my file.txt"
echo bar > "my 'file.txt"
navi # eventually it will print foo/bar
```

```
% test

# Test echo
echo <foo>

# Test fs
cat <file>

$ foo: echo 1 2 3 | tr " " "\n"
$ file: ls
```

**Comment by curio77** [comment]
Ah, so after some further experimentation, I've found out that my issue only crops up if I invoke navi via Ctrl+G.  When I enter `navi` manually, it does indeed work.

I set up navi in fish via `source (navi widget fish)` in a config file (as is the result of the regular navi setup process for fish).  Could you please try reproducing via Ctrl+G?

**Comment by curio77** [comment]
Instead of pressing Ctrl+G, you can of course also run `navi-widget` if the navi plugin is set up, then the effects are the same.  In the `navi-widget` function, there even is indeed that literal `"query"` string.  FWIW, for me, the function looks like this:

```fish
function navi-widget -d "Show cheat sheets"
  begin
    stty sane
    env NAVI_USE_FZF_ALL_INPUTS=true navi --print query (commandline) | perl -pe 'chomp if eof' | read -lz result
    and commandline -- $result
  end
  commandline -f repaint
end
```

**Comment by denisidoro** [comment]
Fixed by #174
